manim/mobject/opengl/opengl_mobject.py

```diff
@@ -2627,13 +2627,16 @@ def set_color_by_xyz_func(
     # For shader data
 
     def refresh_shader_wrapper_id(self):
-        self.shader_wrapper.refresh_id()
+        self.get_shader_wrapper().refresh_id()
         return self
 
     def get_shader_wrapper(self):
         from manim.renderer.shader_wrapper import ShaderWrapper
 
-        self.shader_wrapper = ShaderWrapper(
+        # if hasattr(self, "__shader_wrapper"):
+        # return self.__shader_wrapper
+
```

Comment:
for future reference ?

Reply:
Is it worth turning `shader_wrapper` into a private attribute and potentially breaking code that accessed the attribute directly?

Reply:
That would be "User" code am i right? because i can't find any reference to shader_wrapper inside the code except for those 3 occurences.

But i might be so bold and say no one probably accessed that variable?! Or did you ? but yeah we can leave it like that

Reply:
I didn't, and I'd be surprised if anyone actually did -- but [you never know](https://xkcd.com/1172/) :-)

Reply:
You're right! I forgot about that one!